File: src/FormingVectorModel/FormingVectorModel/Core/Transformations/Invertation.cpp
#include "Invertation.h"
#include "../Objects/Visual/IImage.h"
#include "../Objects/Visual/Color.h"
#include <climits>

void Core::Transformations::Invertation::execute(Objects::Visual::IImage* image)
{
	for (int i = 0; i < image->rows(); i++)
	{
		for (int j = 0; j < image->cols(); j++)
		{
			Objects::Visual::IColor* color = image->get(i, j);
			Objects::Visual::Color new_color(
				UCHAR_MAX - color->red(),
				         UCHAR_MAX - color->green(),
				         UCHAR_MAX - color->blue()
			);
			image->set(i, j, &new_color);
		}
	}
}

File: src/FormingVectorModel/FormingVectorModelLibrary/Core/Transformation/Invertation.cs
using FormingVectorModelLibrary.Core.Objects.Visual;

namespace FormingVectorModelLibrary.Core.Transformation
{
    public class Invertation : IInvertation
    {
        public void Execute(IImage image)
        {
            foreach (IColor color in image)
            {
                color.Red = (byte)(byte.MaxValue - color.Red);
                color.Green = (byte)(byte.MaxValue - color.Green);
                color.Blue = (byte)(byte.MaxValue - color.Blue);
            }
        }
    }
}